pkg/controller/pkg/revision/establisher.go

```diff
@@ -131,30 +133,42 @@ func (e *APIEstablisher) Establish(ctx context.Context, objs []runtime.Object, p
 	return resourceRefs, nil
 }
 
-func (e *APIEstablisher) create(ctx context.Context, obj resource.Object, parent resource.Object, control bool, opts ...client.CreateOption) error {
+func (e *APIEstablisher) create(ctx context.Context, obj runtime.Object, parent resource.Object, control bool, opts ...client.CreateOption) error {
+	o := obj.DeepCopyObject().(resource.Object)
 	ref := meta.AsController(meta.TypedReferenceTo(parent, parent.GetObjectKind().GroupVersionKind()))
 	if !control {
 		ref = meta.AsOwner(meta.TypedReferenceTo(parent, parent.GetObjectKind().GroupVersionKind()))
 	}
 	// Overwrite any owner references on the desired object.
-	obj.SetOwnerReferences([]metav1.OwnerReference{ref})
+	o.SetOwnerReferences([]metav1.OwnerReference{ref})
 	return e.client.Create(ctx, obj, opts...)
 }
 
-func (e *APIEstablisher) update(ctx context.Context, current, desired resource.Object, parent resource.Object, control bool, opts ...client.UpdateOption) error {
+func (e *APIEstablisher) update(ctx context.Context, current, desired runtime.Object, parent resource.Object, control bool, opts ...client.UpdateOption) error {
+	c := current.DeepCopyObject().(resource.Object)
+	d := desired.DeepCopyObject().(resource.Object)
```

Comment:
Ditto the above comment; do we need to deepcopy here?

Reply:
See comment above. The added tests also demonstrate some of this behavior.